com.unity.formats.fbx/Tests/FbxTests/ModelExporterTest.cs

```diff
@@ -1227,5 +1228,72 @@ public void TestMissingMeshExport()
             ModelExporter.ExportObject(filename, cube);
             Assert.IsNotNull(filename);
         }
+
+        [Test]
+        public void TestMaterialScaleAndOffset() 
+        {
+            string matClampGuid = "e6598d8bd228e5940988877e9eddd594";
+            string matRepeatGuid = "f07bd71d1d87a7b41acfc483bf06be3f";
+            
+            var filename = GetRandomFbxFilePath();
+
+            var clampMat = AssetDatabase.LoadAssetAtPath<Material>(AssetDatabase.GUIDToAssetPath(matClampGuid));
+            var cubeClamp = GameObject.CreatePrimitive(PrimitiveType.Cube);
+            cubeClamp.name = "Clamp";
+            cubeClamp.GetComponent<MeshRenderer>().sharedMaterial = clampMat;
+
+            var repeatMat = AssetDatabase.LoadAssetAtPath<Material>(AssetDatabase.GUIDToAssetPath(matClampGuid));
+            var cubeRepeat = GameObject.CreatePrimitive(PrimitiveType.Cube);
+            cubeRepeat.name = "Repeat";
+            cubeRepeat.GetComponent<MeshRenderer>().sharedMaterial = repeatMat;
+
+            var settings = new ExportModelSettingsSerialize();
+            settings.SetEmbedTextures(true);
+            ModelExporter.ExportObjects(filename, new Object[] { cubeClamp, cubeRepeat }, settings);
+
+            var modelImporter = AssetImporter.GetAtPath(filename) as ModelImporter;
+            modelImporter.ExtractTextures(tempUnpackPath);
```

Comment:
To get a random path in assets you can call `GetRandomFileNamePath(extName: "");`.
The base class will automatically delete it on teardown.

Reply:
Nice, that removes the extra Teardown method again.